logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -296,6 +296,24 @@
     end
   end
 
+  context "--config-test" do
+    let(:cli_args) { ["-t", "-e", pipeline_string] }
+
+    context "with a good configuration" do
+      let(:pipeline_string) { "input { } filter { } output { }" }
```

Comment:
I'm on the fence here, it just feels awkward, and for me is an indication that probably this test should be located within other "broader" sections. 

For this change, it works and does the work, so I'm good with it (in a pragmatic way :+1: )